server/src/infra/graphql/query.rs

```diff
@@ -661,18 +665,136 @@ pub fn serialize_attribute(
 }
 
 impl<Handler: BackendHandler> AttributeValue<Handler> {
-    fn from_schema(a: DomainAttributeValue, schema: &DomainAttributeList) -> FieldResult<Self> {
-        match schema.get_attribute_schema(&a.name) {
-            Some(s) => Ok(AttributeValue::<Handler> {
-                attribute: a,
-                schema: AttributeSchema::<Handler> {
-                    schema: s.clone(),
-                    _phantom: std::marker::PhantomData,
-                },
-                _phantom: std::marker::PhantomData,
-            }),
-            None => Err(FieldError::from(format!("Unknown attribute {}", &a.name))),
-        }
+    fn from_schema(a: DomainAttributeValue, schema: &DomainAttributeList) -> Option<Self> {
+        schema
+            .get_attribute_schema(&a.name)
+            .map(|s| AttributeValue::<Handler>::from_domain(a, s.clone()))
```

Comment:
I'm not 100% convinced by the way we are filtering out `!is_visible` fields here, TBH. Wouldn't it be cleaner to do it in `UserRestrictedListerBackendHandler::list_users`, then use that for `Query<Handler>::users` instead of `ReadonlyBackendHandler::list_users`? (:

Reply:
You can't make sense of a (serialized) attribute without the schema, and the schema is restricted to only visible items in `UserRestrictedListerBackendHandler` (access_control.rs:324)